src/libraries/System.Private.CoreLib/src/System/Text/EncodingTable.cs

```diff
@@ -105,18 +106,23 @@ private static int InternalGetCodePageFromName(string name)
         // Return a list of all EncodingInfo objects describing all of our encodings
         internal static EncodingInfo[] GetEncodings()
         {
-            EncodingInfo[] arrayEncodingInfo = new EncodingInfo[s_mappedCodePages.Length];
+            List<EncodingInfo> encodingInfos = new List<EncodingInfo>(s_mappedCodePages.Length);
 
             for (int i = 0; i < s_mappedCodePages.Length; i++)
             {
-                arrayEncodingInfo[i] = new EncodingInfo(
+                if (s_mappedCodePages[i] == Encoding.CodePageUTF7 && !Encoding.IsUTF7EncodingEnabled)
+                {
+                    continue; // skip this entry
+                }
```

Comment:
Should we instead just ensure it's not in s_mappedCodePages?  Then anything else using it won't accidentally include it, and we won't pay the List overhead on every GetEncodings call.

Reply:
Changing that static backing field ends up being fairly intrusive because so many different things depend on it (including the contents of other backing fields), and it results in us having to special-case UTF-7 throughout more methods like `InternalGetCodePageDataItem`. I'll drop the use of `List<T>` here.